pkg/namespace-labeler/labeler.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package namespacelabeler
+
+import (
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/openshift/prom-label-proxy/injectproxy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+// Labeler enables to enforce adding namespace labels to PrometheusRules and to metrics used in them
+type Labeler struct {
+	enforcedNsLabel string
+	excludeList     map[string]map[string]struct{}
+}
+
+// New - creates new Labeler
+// enforcedNsLabel - label name to be enforced for namespace
+// excludeConfig - list of namespace + PrometheusRule names to be excluded while enforcing adding namespace label
+func New(enforcedNsLabel string, excludeConfig []monitoringv1.PrometheusRuleExcludeConfig) *Labeler {
+
+	if enforcedNsLabel == "" || len(excludeConfig) == 0 {
+		return &Labeler{enforcedNsLabel: enforcedNsLabel}
```

Comment:
can we add a comment here that this is essentially a no-op labeler? Just for me in 3months to understand what this is :)